treerec/tskit/tables.c

```diff
@@ -3399,14 +3399,7 @@ tsk_provenance_table_load(tsk_provenance_table_t *self, kastore_t *store)
 /*************************
  * sort_tables
  *************************/
-
-typedef struct {
-    double left;
-    double right;
-    tsk_id_t parent;
-    tsk_id_t child;
-    double time;
-} edge_sort_t;
+//Moved Struct edge_sort_t to tables.h(Sudharshan)
```

Comment:
Again I wonder: what's the long-term strategy here, @petrelharp?  Upon reflection, it seems like it would be a Good Thing for tskit to have some awareness of parallelism; we're presumably not going to be the only tskit client that will want to do these things, and so maybe we ought to plan for the parallelism changes in tskit in SLiM to ultimately get pulled back into tskit itself.  But I guess some of it is C++-based now (does it need to be, or could it be pure C?); is that a total obstacle to it being in tskit, or do you think Jerome might accept a C++ veneer in the tskit sources that would be an optional add-on in some way?  Maybe we need to discuss our approach on this a bit.